### PR TITLE
Content AI in the stories, closes #934

### LINKED ISSUE #934: Add AI assistance to write better stories
https://animina.de/my/stories/new and the edit page of a story look like this:

Bildschirmfoto 2024-08-20 um 21 01 05

A lot of users are not born story tellers and they might have language problems. We should offer them help.

I'd like to add the following functionalities:
- `Correct any spelling, grammar, and punctuation errors.`
- `Improve the story to be funnier.`
- `Improve the story to be more exciting.`
- `Lengthen the story.`
- `Shorten the story.`

Use AI to do it. Feed the original story into the AI and return the output in the input field. We right the prompt in English but we tell the AI to use the language the story was written in.

### PR DESCRIPTION
**Pull request by Kimutai01**
This is the implementation for the chat in the stories both for edit and creating a new one.
I have attached the videos

https://github.com/user-attachments/assets/19c2f387-d1c3-4fdf-a32e-0a4f242a1d21

https://github.com/user-attachments/assets/b8243594-bf67-4457-9680-2fc1793e8fa3

https://github.com/user-attachments/assets/20657cfe-3c2f-4f46-80a4-bf02eed3d4ed

Screenshot 2024-08-22 at 13 45 20

### CODE REVIEW

**Comment by wintermeyer** [comment]
@Kimutai01 I like the functionality. But we have to fiddle around with the GUI a bit. It takes too long for the first word the LLM writes. 

Here's an idea: The moment one presses a button we put the text "Feeding our internal AI with this text. Please wait a second ..." in the text field and add a . (point) every second. To show the user that something happens. The moment the LLM returns first results we put that into the text field.

**Review by wintermeyer — CHANGES_REQUESTED** [review]
@Kimutai01 I like the functionality. But we have to fiddle around with the GUI a bit. It takes too long for the first word the LLM writes.

Here's an idea: The moment one presses a button we put the text "Feeding our internal AI with this text. Please wait a second ..." in the text field and add a . (point) every second. To show the user that something happens. The moment the LLM returns first results we put that into the text field.

**Comment by Kimutai01** [comment]
Aaah..interesting let me try that

File: lib/animina_web/live/story_live.ex [hunk 1]
```diff
@@ -284,6 +303,188 @@ defmodule AniminaWeb.StoryLive do
...
+           ) do
+        {:ok, task} ->
+          new_story =
+            if socket.assigns.story.photo == nil do
+              %{
+                "photo" => %{
+                  "_form_type" => "create",
+                  "_ignored" => "true",
+                  "_persistent_id" => "0",
+                  "user_id" => "1be087b1-f935-4f87-93e0-1129d8dcb56c"
```
Comment (wintermeyer) [inline lib/animina_web/live/story_live.ex]: I didn't dive deep into the code but is this fixed `user_id` intentional?

**Review by wintermeyer — CHANGES_REQUESTED** [review]

File: lib/animina_web/live/story_live.ex (hunk 1 above)
Comment (Kimutai01) [reply lib/animina_web/live/story_live.ex]: @wintermeyer That was a mistake from my end sorry. Let me fix it

**Review by wintermeyer — CHANGES_REQUESTED** [review]
```elixir
  Warnings - please take a look                                                 
┃ 
┃ [W] ↗ There should be no calls to `IO.inspect/1`.
Warning: ┃       lib/animina_web/live/story_live.ex:362:5 #(AniminaWeb.StoryLive.process_story)
┃ [W] ↗ There should be no calls to `IO.inspect/1`.
Warning: ┃       lib/animina_web/live/story_live.ex:361:5 #(AniminaWeb.StoryLive.process_story)
┃ [W] ↗ There should be no calls to `IO.inspect/1`.
Warning: ┃       lib/animina/chat_completion.ex:107:59 #(Animina.ChatCompletion.request_stories)
```

Always run `mix credo` before creating a PR. And when fixing something I requested in a review: Please add new screenshots or screencasts. That is the quickest way for me to get an overview. Thanks!

**Comment by Kimutai01** [comment]
https://github.com/user-attachments/assets/fb9892d8-411e-4daa-aaa8-6de9414559ce


https://github.com/user-attachments/assets/e4e3d99b-023a-497e-a301-e0b7ba99a4bc

**Pull request merged**